arch/armv7/thumb2_disasm/il_thumb2.cpp

```diff
@@ -1475,7 +1512,289 @@ bool GetLowLevelILForNEONInstruction(Architecture* arch, LowLevelILFunction& il,
+				prefix = "s";
+			}
+		}
+		auto regSize = RegisterSizeFromPrefix(prefix);
+		unsigned int d = instr->fields[FIELD_d];
+		unsigned int inc = 1;
+		if(IS_FIELD_PRESENT(instr, FIELD_inc))
+			inc = instr->fields[FIELD_inc];
+		int regs = instr->fields[FIELD_regs];
+		for(int i=0; i<regs; ++i) {
+			if (d+(i*inc) >= 32 && strcmp(prefix, "s") == 0) break;
+			if (i >= 16 && strcmp(prefix, "d") == 0) break;
+			int regIdx = (d + i * inc) % 32;
+			il.AddInstruction(il.SetRegister(regSize, GetRegisterByIndex(regIdx, prefix),
+			                                 il.Load(regSize, il.Add(4, ReadILOperand(il, instr, 0),
+			                                                         il.Const(4, i * regSize)))));
+			}
+		}
+		break;
+	case armv7::ARMV7_VPUSH:
+		{
+		// TODO: Clean this code up...
+		const char* prefix = "d";
+		if (IS_FIELD_PRESENT(instr, FIELD_single_regs)){
+			if (instr->fields[FIELD_single_regs] == 1) {
+				prefix = "s";
+			}
+		}
+		auto regSize = RegisterSizeFromPrefix(prefix);
+		unsigned int d = instr->fields[FIELD_d];
+		unsigned int inc = 1;
+		if(IS_FIELD_PRESENT(instr, FIELD_inc))
+			inc = instr->fields[FIELD_inc];
+		int regs = instr->fields[FIELD_regs];
+		for(int i=0; i<regs; ++i) {
+			if (d+(i*inc) >= 32 && strcmp(prefix, "s") == 0) break;
+			if (i >= 16 && strcmp(prefix, "d") == 0) break;
+			int regIdx = (d + i * inc) % 32;
+			il.AddInstruction(il.Push(regSize, il.Register(regSize, GetRegisterByIndex(regIdx, prefix))));
+		}
+		}
+		break;
+	case armv7::ARMV7_VPOP:
+		{
+		// TODO: Clean this code up...
+		const char* prefix = "d";
+		if (IS_FIELD_PRESENT(instr, FIELD_single_regs)){
+			if (instr->fields[FIELD_single_regs] == 1) {
+				prefix = "s";
+			}
+		}
+		auto regSize = RegisterSizeFromPrefix(prefix);
+		unsigned int d = instr->fields[FIELD_d];
+		unsigned int inc = 1;
+		if(IS_FIELD_PRESENT(instr, FIELD_inc))
+			inc = instr->fields[FIELD_inc];
+		int regs = instr->fields[FIELD_regs];
+		for(int i=0; i<regs; ++i) {
+			if (d+(i*inc) >= 32 && strcmp(prefix, "s") == 0) break;
+			if (i >= 16 && strcmp(prefix, "d") == 0) break;
+			int regIdx = (d + i * inc) % 32;
+			il.AddInstruction(il.SetRegister(regSize, GetRegisterByIndex(regIdx, prefix), il.Pop(regSize)));
+		}
+		}
+		break;
+	case armv7::ARMV7_VLDR:
+		{
+		uint32_t regSize = RegisterSizeFromPrefix(instr->format->operands[1].prefix);
+		if (instr->format->operandCount == 3)
+		{
+			uint32_t reg = GetRegisterByIndex(instr->fields[instr->format->operands[1].field0], instr->format->operands[1].prefix);
+			il.AddInstruction(WriteILOperand(il, instr, 0, il.Load(regSize, GetMemoryAddress(il, instr, 1, regSize, false))));
+			il.AddInstruction(il.SetRegister(regSize, reg, il.Add(regSize, il.Register(regSize, reg), ReadILOperand(il, instr, 2))));
+		}
+		else
+		{
+			il.AddInstruction(WriteILOperand(il, instr, 0, il.Load(regSize, GetMemoryAddress(il, instr, 1, 4))));
+		}
+		}
+		break;
+	case armv7::ARMV7_VSTR:
+		{
```

Comment:
don't indent this `{`, or the closing `}`, but do move the `break;` inside the curly braces so it is indented